lib/elastic_searcher.rb

```diff
@@ -13,24 +25,25 @@ def search
     )
     result.response # ES query is triggered here to allow fallback. avoids lazy loading done in the view
     [nil, result]
-  rescue Faraday::ConnectionFailed, Faraday::TimeoutError, Searchkick::Error, OpenSearch::Transport::Transport::Error => e
-    result = Rubygem.legacy_search(@query).page(@page)
-    [error_msg(e), result]
+  rescue StandardError => e
+    [error_msg(e), nil]
   end
 
   def api_search
     result = Rubygem.searchkick_search(body: search_definition(for_api: true).to_hash, page: @page, per_page: Kaminari.config.default_per_page,
 load: false)
     result.response["hits"]["hits"].pluck("_source")
-  rescue Faraday::ConnectionFailed, Faraday::TimeoutError, Searchkick::Error, OpenSearch::Transport::Transport::Error
-    Rubygem.legacy_search(@query).page(@page)
+  rescue Searchkick::InvalidQueryError
+    raise InvalidQueryError
+  rescue *CONNECTION_ERRORS
+    raise SearchNotAvailableError
   end
 
   def suggestions
     result = Rubygem.searchkick_search(body: suggestions_definition.to_hash, page: @page, per_page: Kaminari.config.default_per_page, load: false)
     result = result.response["suggest"]["completion_suggestion"][0]["options"]
     result.map { |gem| gem["_source"]["name"] }
-  rescue Faraday::ConnectionFailed, Faraday::TimeoutError, Searchkick::Error, OpenSearch::Transport::Transport::Error
+  rescue *CONNECTION_ERRORS
     Array(nil)
```

Comment:
track errors here too

Reply:
I'm trying to keep it 1:1 as much as possible. Errors were not tracked in here before. But I can add it, np.

Reply:
Understand, I'm just concerned about us breaking it somehow and having 0 feedback if we continue to swallow all the errors (which is what caused yesterdays issue, we were falling back on bad queries for a long time and didnt know that)